src/compiler/scala/tools/nsc/settings/ScalaSettings.scala

```diff
@@ -312,6 +321,11 @@ trait ScalaSettings extends AbsScalaSettings
 
   private def replacedByYpredef = "This flag is slated for removal. Its behavior can be achieved with the more generic -Ypredef flag."
 
+  private def requireExclusiveUse(peer: Setting): (Setting) => Unit = { s =>
+    if (peer.isSetByUser)
+      errorFn(s"Flag ${s.name} can not be used in conjunction with flag ${peer.name}")
+  }
+
```

Comment:
This is a bit odd. Is there a word for currying in this way? I think this is the same:
```
def exclusively(peer: Setting)(setting: Setting): Unit = if (peer.isSet) fail(setting)
```